cmd/serve.go

```diff
@@ -20,66 +20,57 @@ var serveCmd = &cobra.Command{
 	Short: "Start configurator as a server and listen for requests",
 	Run: func(cmd *cobra.Command, args []string) {
 		// make sure that we have a token present before trying to make request
-		if config.AccessToken == "" {
-			// TODO: make request to check if request will need token
-
-			// check if OCHAMI_ACCESS_TOKEN env var is set if no access token is provided and use that instead
+		if conf.AccessToken == "" {
+			// check if ACCESS_TOKEN env var is set if no access token is provided and use that instead
 			accessToken := os.Getenv("ACCESS_TOKEN")
 			if accessToken != "" {
-				config.AccessToken = accessToken
+				conf.AccessToken = accessToken
 			} else {
-				// TODO: try and fetch token first if it is needed
 				if verbose {
-					fmt.Printf("No token found. Attempting to generate config without one...\n")
+					fmt.Printf("No token found. Continuing without one...\n")
 				}
```

Comment:
Any reason we are mixing `if verbose { fmt.printf(...) }` with using `log` (line 39 below)? We probably want to restructure this to use something like `log.Warn()` or `log.Info()` repository-wide instead and set the log-level.

Reply:
Yeah, I think this is just something I missed. I'm going to change these.

Reply:
I found some additional files with printfs that should potentially be log messages:
```
./cmd/generate.go:39:					fmt.Printf("No token found. Attempting to generate conf without one...\n")
./cmd/generate.go:56:			fmt.Printf("%v\n", string(b))
./cmd/generate.go:64:				fmt.Printf("no plugin path specified")
./cmd/config.go:23:				fmt.Printf("file or directory exists\n")
./cmd/fetch.go:36:					fmt.Printf("No token found. Attempting to generate config without one...\n")
./cmd/serve.go:41:			fmt.Printf("%v\n", string(b))
./cmd/root.go:37:		fmt.Fprintln(os.Stderr, err)
./pkg/generator/generator.go:67:				fmt.Println(err)
./pkg/generator/generator.go:141:			fmt.Printf("-- found plugin '%s'\n", gen.GetName())
./pkg/generator/dhcpd.go:54:		computeNodes += fmt.Sprintf("host %s { hardware ethernet %s; fixed-address %s} ", eth.ComponentId, eth.MacAddress, eth.IpAddresses[0])
./pkg/generator/dhcpd.go:59:		fmt.Printf("")
./pkg/client/smd.go:60:		fmt.Printf("Ethernet Interfaces: %v\n", string(bytes))
./pkg/client/smd.go:102:		fmt.Printf("Components: %v\n", string(bytes))
./pkg/client/smd.go:141:		fmt.Printf("Redfish endpoints: %v\n", string(b))
```

Reply:
Leftovers:
https://github.com/OpenCHAMI/configurator/blob/d2b6178350b2bdafb4e0083fcda5ea55e1b1cf4f/cmd/fetch.go#L36
https://github.com/OpenCHAMI/configurator/blob/d2b6178350b2bdafb4e0083fcda5ea55e1b1cf4f/pkg/generator/dhcpd.go#L58-L60
https://github.com/OpenCHAMI/configurator/blob/d2b6178350b2bdafb4e0083fcda5ea55e1b1cf4f/pkg/client/smd.go#L59-L61
https://github.com/OpenCHAMI/configurator/blob/d2b6178350b2bdafb4e0083fcda5ea55e1b1cf4f/pkg/client/smd.go#L101-L103
https://github.com/OpenCHAMI/configurator/blob/d2b6178350b2bdafb4e0083fcda5ea55e1b1cf4f/pkg/client/smd.go#L140-L142
https://github.com/OpenCHAMI/configurator/blob/d2b6178350b2bdafb4e0083fcda5ea55e1b1cf4f/pkg/generator/generator.go#L145
https://github.com/OpenCHAMI/configurator/blob/d2b6178350b2bdafb4e0083fcda5ea55e1b1cf4f/cmd/config.go#L23 (not sure if logging is set up at this point, either way the name of the file that exists should be specified here to be more informative)

Reply:
I changed this to output the error which includes that information.

Reply:
Looks good.